Sources/Orbit/Components/HorizontalScroll.swift

```diff
@@ -23,301 +28,521 @@ public enum HorizontalScrollItemHeight {
+        let maxOffset = -contentSize.width + scrollViewWidth
+        let isContentBiggerThanScrollView = contentSize.width > scrollViewWidth
 
-            HorizontalScroll(spacing: .large, itemWidth: .ratio(0.33)) {
-                Color.greenLight.frame(height: 10)
-                Color.greenLight.frame(height: 30)
-                Color.greenLight.frame(height: 50)
-                Color.greenLight.frame(height: 20)
-            }
-            .border(Color.cloudDark)
+        guard isContentBiggerThanScrollView else { return }
+
+        if currentOffset > scrollViewWidth / 5 {
+            stopSnappingAnimation()
+        } else  if currentOffset < maxOffset - scrollViewWidth / 5 {
+            stopSnappingAnimation()
+        } else {
+            snappingOffset = scrollSnappingOffset(currentOffset: currentOffset)
+        }
+    }
+
+    private func scrollSnappingOffset(currentOffset: CGFloat) -> CGFloat {
+        if currentOffset > 0 {
+            return adjustedSnappingOffsetOverBounds(offsetOverBounds: currentOffset)
+        } else if currentOffset < maxOffset {
+            return maxOffset - adjustedSnappingOffsetOverBounds(offsetOverBounds: abs(currentOffset - maxOffset))
+        } else {
+            return currentOffset
         }
-        .background(Color.redLight.frame(width: 1).padding(.leading, .xSmall), alignment: .leading)
-        .background(Color.redLight.frame(width: 1).padding(.trailing, .xSmall), alignment: .trailing)
-        .previewDisplayName("W - ratios, H - intrinsic")
     }
 
-    static var simpleCustom: some View {
-        HorizontalScroll(itemWidth: .custom(30), itemHeight: .custom(30)) {
-            Color.greenLight
-            Color.greenLight
-            Color.greenLight
+    private func stopSnappingAnimation() {
+        withAnimation(.spring()) {
+            snappingOffset = getSnappingOffsetInBounds(offset: snappingOffset)
         }
-        .border(Color.cloudDark)
-        .previewDisplayName("W - custom, H - custom")
+        isAnimating = false
+        timer.connect().cancel()
+        lastOffset = snappingOffset
     }
 
-    static var ratioWidthIntrinsicHeight: some View {
-        HorizontalScroll {
-            intrinsicContent
+    private func updateOffset(gesture: DragGesture.Value) {
+        if scrollingInProgress == false {
+            stopSnappingAnimation()
+        }
+
+        scrollingInProgress = true
+
+        let currentOffset = gesture.translation.width + lastOffset
 
-            intrinsicContent {
-                Color.greenLight
-                    .frame(width: 100, height: 150)
+        guard isContentBiggerThanScrollView else { return }
+
+        snappingOffset = scrollSnappingOffset(currentOffset: currentOffset)
+    }
+
+    private func gestureEnded(gesture: DragGesture.Value) {
+        scrollingInProgress = false
+
+        guard let itemWidth = resolvedItemWidth else { return }
+
+        guard isContentBiggerThanScrollView else { return }
```

Comment:
`guard`s could be combined.

Reply:
The implementation changed a bit, should be updated.